src/components/forms/PractitionerForm/UserIdSelect/index.tsx

```diff
@@ -25,6 +25,10 @@ export interface Props {
   showPractitioners: boolean /** show users that are already mapped to a practitioner */;
   className: string;
   ReactSelectDefaultValue: Option;
+  /** if we are only interested in the userName -> affects how a single option
```

Comment:
Is it possible to just pick one of these to use instead of making it interchangeable?

Reply:
yeah, though in that case we would have the userName as the label,even if we intend to use it as the value

Reply:
> Additionally, if as you say we are using the username to filter, does it mean that we filter using the label of the react-select option

Reply:
What do we use the `id` for?

Reply:
its used in the PractionerForm when creating a practitioner @moshthepitt

Reply:
@p-netm I mean when doing user filters like here, how is the id used?

Reply:
its not being used for filtering in this useCase, but I would rather pass in a different prop than have different schema in the data passed to the callback, thoughts?

Reply:
@p-netm  Could you elaborate on what you mean?  What is causing the differences in schema?

Reply:
hmm,.. stepping back here:
> Is it possible to just pick one of these to use instead of making it interchangeable?
how did mean, I though you referred to how the selected option is passed back to the callback, where in this case, you would call the callback with just the value you need which would be a string

Reply:
by the way; I removed the use of the userId from the filtering functionality, the passed back option has the username as both the value of `label` and `value`

Reply:
I meant that if we are not using the id at all when filtering, why do we need to have a mechanism to conditionally use either id or username as value?  Or is it because this is shared between filtering and the practitioner form?

Reply:
ooh ok. yeah, this is a shared component.